src/main/kotlin/de/bund/digitalservice/useid/widget/WidgetController.kt

```diff
@@ -149,13 +150,17 @@ class WidgetController(
        If major version returns empty or somehow the user agent parser cannot parse the version properly,
        we will show to users the widget
     */
-    private fun safelyGetMajorOSVersion(majorVersion: String?): Int {
-        val fallbackOSVersion = 20
-
+    private fun safelyGetMajorOSVersion(majorVersion: String?): Int? {
         return try {
             Integer.parseInt(majorVersion)
         } catch (exception: NumberFormatException) {
-            fallbackOSVersion
+            null
         }
     }
+
+    private fun hasSupportedMajorOSVersion(parsedUserAgent: Client, osFamily: String, supportedMajorVersion: Int): Boolean {
+        return parsedUserAgent.os.family == osFamily &&
+            parsedUserAgent.os.major != null &&
+            safelyGetMajorOSVersion(parsedUserAgent.os.major)!! <= supportedMajorVersion
```

Comment:
This needs to be the other way round, right? We support versions "from 14" / "from 8" upwards and not downwards.

Reply:
The name of the method is misleading. The method is used to set the boolean values `incompatibleIOSVersion` and  `incompatibleAndroidVersion`. So the method  returning `true` means that the version is incompatible. Please rename the method or reverse the logic.

Reply:
🙃